CROSS-CUTTING-CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Creating a development environment
+
+When cross cutting changes are needed, it is best to work in a personal branch, but would need to replicate this to each submodule impacted.
+
+```
+$ git checkout -b CASSANDRA-<number>
+# Change the URL to point to your fork
+$ git submodule set-url accord "git@github.com:$USER/cassandra-accord.git"
+# Change the branch to your development branch
+$ (branch="$(git rev-parse --abbrev-ref HEAD)"; cd accord && git checkout -b "$branch")
+```
+
+To better automate this a utility script exists, so can do
+
+```
+$ dev-support/development-switch.sh --jira CASSANDRA-<number>
+```
+
```

Comment:
At this point in the narrative, a couple things we could do:

1.) Make it clear for readers who aren't familiar w/ submodules that any commits they make to Accord will be visible to C* immediately, and local testing will be possible. (Let's say this ends the "Local Development" section.)

2.) We could start a new section, let's call it "Remote Testing", that covers `bump-accord.sh`, which, because of the pre-commit hook, starts pushing things to remotes. (Then pushing the C* bits makes remote testing on Circle or wherever possible, etc.)

Reply:
> that any commits they make to Accord will be visible to C* immediately

I guess I assumed that by us having it in the repo... why have it in the repo if you can't see it?

Do we need to call out `Remote Testing`?  The defaults are setup to work in Circle/Jenkins out of the box, so there is nothing to actually do to enable; you push like normal and CI runs like normal.

>  because of the pre-commit hook, starts pushing things to remotes

I feel like that is tangental, we push to avoid bad commits to Cassandra... If we don't push you could create a commit that points to a bad accord SHA.  We *could* document this, but is this something people need to think about?  There is a very specific edge case I see and that is if an committer is committing to `trunk`... they might not know that they try to push that change to `apache` (unless they updated to their repo)... since people tend to work in feature branches, I am not sure how much of a concern this should be